include/device.h

```diff
@@ -85,6 +85,26 @@ extern "C" {
 	DEVICE_AND_API_INIT(dev_name, drv_name, init_fn,\
 	data, cfg_info, level, prio, NULL)
 
+/**
+ * @def DEVICE_EXEC_FOR_ALL_INSTANCES
```

Comment:
I think now that #23245 is in it would be cleaner to:

- put this feature into devicetree.h instead of device.h
- make it something like `DT_INST_FOREACH(inst_macro)`, where `inst_macro` just takes 1 argument: the instance number
- `DT_INST_FOREACH` could probably use `DT_NUM_INST(DT_DRV_COMPAT)` to figure out how many instances there are; if not, we can make it possible some other way

Reply:
I changed it.
Now it uses the newly introduced DT_NUM_INST instead of using a while-loop to detect the old-style DT_INST_idx_compat nodes.
Also as suggested now in devicetree.h.